src/ckb/hash_algorithm.rs

```diff
@@ -0,0 +1,79 @@
+use bitcoin::hashes::{sha256::Hash as Sha256, Hash as _};
+use ckb_hash::blake2b_256;
+use ckb_types::packed;
+use serde::{Deserialize, Serialize};
+use thiserror::Error;
+
+#[repr(u8)]
+#[derive(Copy, Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(rename_all = "snake_case")]
```

Comment:
Side note: there are many other places where we want to deserialize a string into a enum. Some of they are case insensitive (e.g. https://github.com/nervosnetwork/cfn-node/blob/c69f8121f9a542ddeae4d38e467234024ec92c8f/src/ckb_chain/config.rs#L108-L114 ), some of they are case sensitive, some of they just use the default serde config (e.g. https://github.com/nervosnetwork/cfn-node/blob/c69f8121f9a542ddeae4d38e467234024ec92c8f/src/ckb/config.rs#L127-L129). This can create confusion. We may want to unify all these serialization/deserialization procedures.